camayoc/tests/qpc/api/v1/scanjobs/test_pause_cancel_restart.py

```diff
@@ -65,7 +65,7 @@ def test_restart(shared_client, cleanup, source_type):
         6) Assert that the scan completes
     :expectedresults: A restarted scan completes
     """
-    scn = util.prepare_scan(source_type, cleanup)
+    scn = util.prepare_scan(source_type, data_provider)
```

Comment:
I admit this is one use case I have missed - sometimes you want an object, but you have certain requirements about its *dependencies*. It's not very clear how DataProvider could fill this, especially while maintaining backwards compatibility. 

Right now your only option is to ask for a first object in dependency chain that you have requirements for, and then construct the rest manually. This requires you to still call `QPCModel.create()` and `cleanup()`. You can hide this in helper function, like here, but it does _feel_ like something DataProvider should provide.